### PR TITLE
Fix crash when closing last tab

### PR DESCRIPTION
**Pull request by fghzxm**
## Summary of the Pull Request
This commit fixes crashes that happen when the last tab is being closed.

## References
None.

## PR Checklist
* [ ] Closes #xxx
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Detailed Description of the Pull Request / Additional comments
Commit bc69d1a99a8921b5088c035096e659b9e0a1d50e adds code that closes
the window when the last tab is being closed, but code that should only
be executed when closing one of multiple open tabs continues to be
executed.  This leads to App::_RemoveTabViewItem trying to call
TabView::SelectedIndex(1) when there is only 1 tab remaining in the
TabView, resulting in an argument exception.

### CODE REVIEW

File: src/cascadia/TerminalApp/App.cpp [hunk 1]
```diff
@@ -846,25 +846,27 @@ namespace winrt::TerminalApp::implementation
...
+            }
+            _tabView.Items().RemoveAt(removingTabIndex);
         }
 
         // Removing the tab from the collection will destroy its control and disconnect its connection.
-        _tabs.erase(_tabs.begin() + tabIndexFromControl);
-        _tabView.Items().RemoveAt(tabIndexFromControl);
-
-        // ensure tabs and focus is sync
-        _tabView.SelectedIndex(tabIndexFromControl > 0 ? tabIndexFromControl - 1 : 0);
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/App.cpp]: This got removed. But I'm noticing it's similar to line 863. Should we be removing this line and updating/replacing line 863 with this?

File: src/cascadia/TerminalApp/App.cpp [hunk 2]
```diff
@@ -846,25 +846,27 @@ namespace winrt::TerminalApp::implementation
...
+        else
         {
-            _tabView.SelectedIndex((tabIndexFromControl > 0) ? tabIndexFromControl - 1 : 1);
+            // If we are closing the tab we are on, flee to an adjacent tab first!
+            // We go to the tab on the right if we are on the 0th tab, or the one on the left otherwise.
+            if (removingTabIndex == _GetFocusedTabIndex())
+            {
+                _tabView.SelectedIndex(removingTabIndex > 0 ? removingTabIndex - 1 : 1);
+            }
+            _tabView.Items().RemoveAt(removingTabIndex);
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/App.cpp]: Should this be moved down with line 869 like it was before? Or does closing the window (line 855) basically handle the logic of removing the XAML TabView Control?

File: src/cascadia/TerminalApp/App.cpp (hunk 1 above)
Comment (fghzxm) [reply src/cascadia/TerminalApp/App.cpp]: From my experiments, line 867 is unnecessary.  The `SelectedIndex` property is maintained correctly by the `TabView` component when a tab is closed.

**Comment by fghzxm** [comment]
I'm closing this PR for now since PR #818 seems to obsolete this.  However I'm still monitoring https://github.com/microsoft/microsoft-ui-xaml/pull/629 which I believe very likely contains the bug of `TabView::OnItemsChanged` failing to maintain `SelectedIndex` correctly.  In fact, the extra logic PR #818 introduces really looks like they should be part of `TabView::OnItemsChanged` itself.

**Pull request closed without merging**